compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -274,171 +288,210 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     }
   }
 
-  /** Create a threadsafe lazy accessor equivalent to such code
-    * ```
-    * def methodSymbol(): Int = {
-    *   while (true) {
-    *     val flag = LazyVals.get(this, bitmap_offset)
-    *     val state = LazyVals.STATE(flag, <field-id>)
-    *
-    *     if (state == <state-3>) {
-    *       return value_0
-    *     } else if (state == <state-0>) {
-    *       if (LazyVals.CAS(this, bitmap_offset, flag, <state-1>, <field-id>)) {
-    *         try {
-    *           val result = <RHS>
-    *           value_0 = result
-    *           nullable = null
-    *           LazyVals.setFlag(this, bitmap_offset, <state-3>, <field-id>)
-    *           return result
-    *         }
-    *         catch {
-    *           case ex =>
-    *             LazyVals.setFlag(this, bitmap_offset, <state-0>, <field-id>)
-    *             throw ex
-    *         }
-    *       }
-    *     } else /* if (state == <state-1> || state == <state-2>) */ {
-    *       LazyVals.wait4Notification(this, bitmap_offset, flag, <field-id>)
-    *     }
-    *   }
-    * }
-    * ```
-    */
-  def mkThreadSafeDef(methodSymbol: TermSymbol,
+  /**
+   * Create a threadsafe lazy accessor equivalent to the following code:
+   * ```
+   * private @volatile var _x: AnyRef = null
+   * @tailrec def x: A =
+   *    _x match
+   *    case null =>
+   *        if CAS(_x, null, Evaluating) then
```

Comment:
The indentation within this comment is not regular. Please make it follow 2 spaces like everywhere else.